controllers/machine_controller_phases.go

```diff
@@ -78,6 +78,13 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, m *clusterv1.
 	obj, err := external.Get(r.Client, ref, m.Namespace)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
+			if m.Status.InfrastructureReady && m.Status.GetTypedPhase() == clusterv1.MachinePhaseRunning {
+				// Infra object went missing after the machine was up and running
+				klog.Infof("Infrastructure object %v %q went missing. Updating Machine Status", ref.GroupVersionKind(), ref.Name)
+				machineStatusError := capierrors.MachineStatusError("Invalid Configuration")
+				m.Status.ErrorReason = &machineStatusError
+				m.Status.ErrorMessage = pointer.StringPtr("Invalid Configuration")
```

Comment:
This should be descriptive and include the GVK (GroupVersionKind) and name of the resource that went missing.

Reply:
ok. For both ```ErrorReason``` and ```ErrorMessage``` ??